cirq/sim/density_matrix_simulator.py

```diff
@@ -290,8 +290,10 @@ def keep(potential_op: ops.Operation) -> bool:
                             indices,
                             qid_shape=qid_shape,
                             out=state.tensor)
-                        corrected = [bit ^ mask for bit, mask in
-                                     zip(bits, invert_mask)]
+                        corrected = [
+                            bit ^ (bit < 2 and mask)
```

Comment:
Where does the `2` come from? This looks like it can't possibly be correct.

Reply:
If (the measured value is 0 or 1) and invert_mask, then toggle.